packages/frontpage/lib/data/db/comment.ts

```diff
@@ -232,3 +242,28 @@ export function shouldHideComment(comment: CommentModel) {
     comment.children.length === 0
   );
 }
+
+type ModerateCommentInput = {
+  rkey: string;
+  authorDid: DID;
+  cid: string;
+  hide: boolean;
+};
+export async function moderateComment({
+  rkey,
+  authorDid,
+  cid,
+  hide,
+}: ModerateCommentInput) {
+  console.log(`Moderating comment, setting hidden to ${hide}`);
+  await db
```

Comment:
This (along with all other DAL methods) should have authentication, and in the case of the things that require admin, authorisation checks.